rca/shortcourses/tests/test_access_planit.py

```diff
@@ -0,0 +1,261 @@
+            ShortCoursePage.objects.create(
+                title=f"Short course should not save", path=100, depth="001"
+            )
+        with self.assertRaises(TypeError):
+            ShortCoursePage.objects.create(
+                title=f"Short course should not save",
+                path=100,
+                depth="001",
+                access_planit_course_id="course id cannot be a string",
+            )
+
+    def test_management_command_fetch_data(self):
+        """ Test that xml with no dates doesn't go in the cache """
+        for i in range(5):
+            ShortCoursePage.objects.create(
+                title=f"Short course {i}",
+                path=str(i),
+                depth="001",
+                access_planit_course_id=i,
+            )
+        call_command("fetch_access_planit_data")
+        for i in range(5):
+            cache_key = f"short_course_{i}"
+            self.assertEqual(cache.get(cache_key), None)
+
+    def test_cache_data_for_non_valid_course_id(self):
+        """ Technically, you can add any integer as a course ID and some will
+        not fetch data"""
+        data = AccessPlanitXML(course_id=1).get_data()
+        self.assertEqual(data, None)
+
+    @mock.patch(
+        "rca.shortcourses.access_planit.AccessPlanitXML.fetch_data_from_xml",
+        side_effect=mocked_fetch_data_from_xml,
+    )
+    def test_management_command_fetch_with_example_data(
+        self, mocked_fetch_data_from_xml
+    ):
+        """ Test that the example xml goes in the cache as expected """
+        for i in range(5):
+            ShortCoursePage.objects.create(
+                title=f"Short course {i}",
+                path=str(i),
+                depth="001",
+                access_planit_course_id=i,
+            )
+        args = []
+        opts = {}
+        call_command("fetch_access_planit_data", *args, **opts)
+        for i in range(5):
+            cache_key = f"short_course_{i}"
+            self.assertEqual(cache.get(cache_key), self.expected_data)
+
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_page_renders_with_timeout(self, mock_get):
+        logging.disable(logging.CRITICAL)
+        """ If there is a timeout for the xml request when the page loads,
+        ensure the page still renders with the empty data"""
+        home_page = HomePage.objects.first()
+        mock_get.side_effect = Timeout
+        short_course_page = ShortCoursePage(
+            title="Short course title", slug="short-course", access_planit_course_id="1"
+        )
+        home_page.add_child(instance=short_course_page)
+        response = self.client.get("/short-course/")
+        self.assertTemplateUsed(
+            response, "patterns/pages/shortcourses/short_course.html"
+        )
+        self.assertContains(response, "Short course title")
+        self.assertEqual(
+            response.render().status_code, 200
+        )  # will render 404 if there is an failure
```

Comment:
Which failure will raise a 404?

Reply:
I'll remove this - was just my notes when writing the tests